huaweicloud/services/cts/resource_huaweicloud_cts_data_tracker.go

```diff
@@ -183,54 +186,49 @@ func buildDeleteTagOpt(taglist *[]cts.Tags, id string) *cts.BatchDeleteResourceT
 
 	return &tagOpt
 }
-
-func buildDataBucketOpts(d *schema.ResourceData) *cts.DataBucket {
-	dataBucketCfg := cts.DataBucket{
-		DataBucketName: utils.String(d.Get("data_bucket").(string)),
+func buildDataBucketOpts(d *schema.ResourceData) map[string]interface{} {
+	dataBucketCfg := map[string]interface{}{
+		"data_bucket_name": d.Get("data_bucket").(string),
 	}
 
-	var rawOperations []interface{}
-	if v, ok := d.GetOk("data_operation"); ok {
-		rawOperations = v.([]interface{})
-	} else {
+	rawOperations, ok := d.GetOk("data_operation")
+	if !ok {
 		rawOperations = []interface{}{"READ", "WRITE"}
 	}
 
-	operations := make([]cts.DataBucketDataEvent, len(rawOperations))
-	dataEvents := cts.GetDataBucketDataEventEnum()
-	for i, raw := range rawOperations {
+	operations := make([]interface{}, 0)
+	for _, raw := range rawOperations.([]interface{}) {
 		if op, ok := raw.(string); ok {
-			if op == "READ" {
-				operations[i] = dataEvents.READ
-			} else if op == "WRITE" {
-				operations[i] = dataEvents.WRITE
+			switch op {
+			case "READ", "WRITE":
+				operations = append(operations, op)
 			}
 		}
 	}
-	dataBucketCfg.DataEvent = &operations
+	dataBucketCfg["data_event"] = operations
 
-	return &dataBucketCfg
+	return dataBucketCfg
 }
 
-func buildTransferBucketOpts(d *schema.ResourceData) *cts.TrackerObsInfo {
-	transferCfg := cts.TrackerObsInfo{
-		BucketName:      utils.String(d.Get("bucket_name").(string)),
-		FilePrefixName:  utils.String(d.Get("file_prefix").(string)),
-		IsSortByService: utils.Bool(d.Get("is_sort_by_service").(bool)),
+func buildTransferBucketOpts(d *schema.ResourceData) map[string]interface{} {
+	transferCfg := map[string]interface{}{
+		"bucket_name":        utils.ValueIgnoreEmpty(d.Get("bucket_name").(string)),
+		"file_prefix_name":   utils.ValueIgnoreEmpty(d.Get("file_prefix").(string)),
+		"is_sort_by_service": d.Get("is_sort_by_service").(bool),
 	}
 	if v, ok := d.GetOk("compress_type"); ok {
-		compressType := cts.GetTrackerObsInfoCompressTypeEnum().GZIP
+		compressType := "gzip"
 		if v.(string) != "gzip" {
-			compressType = cts.GetTrackerObsInfoCompressTypeEnum().JSON
+			compressType = "json"
 		}
-		transferCfg.CompressType = &compressType
+		transferCfg["compress_type"] = compressType
```

Comment:
`compress_type` only support **gzip** and **json**.
```suggestion
		transferCfg["compress_type"] = v
```